apm-agent-core/src/main/java/co/elastic/apm/agent/tracemethods/MethodMatcher.java

```diff
@@ -32,6 +33,7 @@
 import static co.elastic.apm.agent.matcher.AnnotationMatcher.annotationMatcher;
 import static co.elastic.apm.agent.matcher.WildcardMatcher.caseSensitiveMatcher;
 
+@GlobalState
```

Comment:
[question] If I understand it correctly, we only need global state here as it relates to configuration like `MethodMatcherValueConverter`, and thus needs to be loaded in the main agent classloader and not in the plugin classloader, is that correct ? If so, maybe a small comment could make that slightly more obvious.

Reply:
Correct. Agree on adding an explanatory comment.